source/elements/oneMKL/source/domains/data_fitting/splines.rst

```diff
@@ -0,0 +1,177 @@
+.. _splines:
+
+Splines
+=======
+
+.. contents::
+    :local:
+    :depth: 1
+
+Header File
+-----------
+
+.. code:: cpp
+
+  #include<oneapi/mkl/experimental/data_fitting.hpp>
+
+Namespace
+---------
+
+.. code:: cpp
+
+  oneapi::mkl::experimental::data_fitiing
+
+Common API for All Spline Types
+-------------------------------
+
+.. code:: cpp
+
+  template <
+    typename FpType,
+    typename SplineType,
+    int Dimensions = 1>
+  class spline {
+  public:
+      using value_type = FpType;
+      using spline_type = SplineType;
+
+      spline(
+        const sycl::queue& q,
+        std::int64_t ny = 1);
+
+      spline(
+        const sycl::device& dev,
+        const sycl::context& ctx,
+        std::int64_t ny = 1);
+
+      ~spline();
+
+      spline(const spline<FpType, SplineType, Dimensions>& other) = delete;
+      spline(spline<FpType, SplineType, Dimensions>&& other) = delete;
+      spline<FpType, SplineType, Dimensions>& operator=(
+          const spline<FpType, SplineType, Dimensions>& other) = delete;
+      spline<FpType, SplineType, Dimensions>& operator=(
+          spline<FpType, SplineType, Dimensions>&& other) = delete;
+
+      spline& set_partitions(
+        FpType* input_data,
+        std::int64_t nx,
+        partition_hint PartitionHint = partition_hint::non_uniform);
+
+      spline& set_function_values(
+        FpType* input_data,
+        function_hint FunctionHint = storage_hint::row_major);
+
+      spline& set_coefficients(
+        FpType* data,
+        coefficient_hint CoeffHint = storage_hint::row_major);
+
+      bool is_initialized() const;
+
+      std::int64_t get_required_coeffs_size() const;
+
+      sycl::event construct(const std::vector<sycl::event>& dependencies = {});
+  };
+
+An instance of ``spline<T, ST, N>`` create the ``N``-dimensional spline
+that operates with the ``T`` data type. ``ST`` is a type of spline.
+``T`` can only be ``float`` or ``double``.
+
+.. list-table::
+   :header-rows: 1
+
+   * - Constructor
+     - Description
+   * - ``spline(const sycl::queue& q, std::int64_t ny = 1);``
+     - Create an object with the ``q`` SYCL queue and ``ny`` number of functions.
+   * - ``spline(const sycl::device& dev, const sycl::context& ctx, std::int64_t ny = 1);``
+     - Create an object using the ``dev`` SYCL device, the ``ctx`` context
+       and ``ny`` number of functions.
+
+.. list-table::
+   :header-rows: 1
+
+   * - Member function
+     - Description
+   * - ``spline& set_partitions(FpType* input_data, std::int64_t nx, partition_hint PartitionHint = partition_hint::non_uniform);``
+     - Set partition values that are specified by the ``input_data`` memory pointer and ``nx`` partition values.
+       Users can provide ``PartitionHint`` to specify the layout of data.
+       Default layout is ``non_uniform``.
+       If ``uniform`` is specified, ``nx`` must equals to ``2`` and
+       ``input_data`` must contain only 2 values the left and the right borders of partition.
```

Comment:
If ``uniform`` is specified, ``nx`` must be equal to ``2`` and ``input_data`` must contain only 2 values corresponding to the left and the right borders of the sub-interval defined by the partition.